readme.md

```diff
@@ -0,0 +1,226 @@
+plugins that work with markdown.
+You can choose from the 150+ plugins that already exist or make your own.
+
+See [remark][] for info on what the remark ecosystem is.
+
+## When should I use this?
+
+You can use this extension if you want to check (lint) and format markdown
+files from within your editor.
+
+To configure this extension, you define your preferred markdown style in a
+configuration file (`.remarkrc`, `.remarkrc.js`, etc. or in `package.json`).
+This file is picked up by `vscode-remark` and other tools (useful for
+contributors that don’t use VS Code).
+
+The configuration file is also used by [`remark-cli`][remark-cli], which is
+recommended to be used alongside `vscode-remark`, as an npm script and/or in
+CI, to enforce the markdown style.
+
+## Install
+
+[Get it from the VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=xxxx).
+Install through VS Code extensions.
+Search for `remark`.
+
+It can also be installed in VS Code: Launch VS Code Quick Open
+(<kbd>Ctrl</kbd> + <kbd>P</kbd>), paste the following command, and press enter.
+
+```txt
+ext install unifiedjs.vscode-remark
+```
+
+See the
+[extension installation guide](https://code.visualstudio.com/docs/editor/extension-gallery)
+for details.
+
+## Use
+
+This example enables VS Code users that have this extension installed to see
+warnings about markdown style in their editor and the `Problems` pane and to
+format markdown from a command.
+
+Make sure that this extension is installed and then install the CLI and
+plugins:
+
+```sh
+npm install remark-cli remark-preset-lint-recommended remark-preset-lint-consistent remark-toc
+```
+
+Then, add a `remarkConfig` to your `package.json` to configure remark:
+
+```js
+  /* … */
+  "remarkConfig": {
+    "settings": {
+      "bullet": "*", // Use `*` for list item bullets (default)
+      // See <https://github.com/remarkjs/remark/tree/main/packages/remark-stringify> for more options.
+    },
+    "plugins": [
+      "remark-preset-lint-consistent", // Check that markdown is consistent.
+      "remark-preset-lint-recommended", // Few recommended rules.
+      [
+        // Generate a table of contents in `## Contents`
+        "remark-toc",
+        {
+          "heading": "contents"
+        }
+      ]
+    ]
+  },
+  /* … */
+```
+
+> 👉 **Note**: you must remove the comments in the above examples when
+> copy/pasting them, as comments are not supported in `package.json` files.
+
+Now, you can open markdown files in your project, and you’ll see squiggly lines
+and warnings in the `Problems` pane.
+Here’s an example that should produce problems you can use to verify:
+
+```markdown
+1) Hello, _Jupiter_ and *Neptune*!
+```
+
+## Formatting
+
+This extension can format markdown files.
+
+To format a file, first press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>.
+Search for and select `Format Document With…`.
+Now select `remark`.
```

Comment:
```suggestion
To format a file, pull up the command pallete (<kbd>Ctrl</kbd> +
<kbd>Shift</kbd> + <kbd>P</kbd>) and choose: `remark: beautify markdown code`
```

Why not recommend one step less?
Inspired by the wording used in https://marketplace.visualstudio.com/items?itemName=samverschueren.linter-xo.
Note: this command (`Remark: Beautify markdown code`) is based on the old extension, what is it now? I don’t really like the wording of `Beautify` and feel like `code` is superfluous?

Reply:
The instructions you’re proposing are simply wrong. They describe the old behavior of the extension, which is to register the command: `Remark: Beautify markdown code`. The new behavior is to register a formatter, which can be used as is currently described or can be set as the default formatter as described below.

Reply:
Why is the command removed?

---

Even if now 3 steps are required, the wording from `linter-xo` can be used as inspiration:

```suggestion
To format a file, pull up the command pallete (<kbd>Ctrl</kbd> +
<kbd>Shift</kbd> + <kbd>P</kbd>), choose `Format Document With…`, and select
`remark`.
```

Reply:
> Why is the command removed?

Because it has been replaced by a formatter.